src/hooks/core/chat/useChatUtilsHook.ts

```diff
@@ -8,18 +8,26 @@ import {atom, useRecoilState} from 'recoil';
 import AnonymousMessageRepo from '../../../service/repo/anonymousMessageRepo';
 import SignedMessageRepo from '../../../service/repo/signedMessageRepo';
 import UseChatUtilsHook from '../../../../types/hooks/screens/useChatUtilsHook.types';
+import UserSchema from '../../../database/schema/UserSchema';
 import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
 import {ANON_PM, GROUP_INFO} from '../constant';
-import {ChannelList} from '../../../../types/database/schema/ChannelList.types';
+import {
+  BetterSocialChannelType,
+  ChannelList
+} from '../../../../types/database/schema/ChannelList.types';
 import {ChannelTypeEnum} from '../../../../types/repo/SignedMessageRepo/SignedPostNotificationData';
 import {Context} from '../../../context';
 import {PostNotificationChannelList} from '../../../../types/database/schema/PostNotificationChannelList.types';
-import {convertTopicNameToTopicPageScreenParam} from '../../../utils/string/StringUtils';
+import {
+  convertTopicNameToTopicPageScreenParam,
+  getChannelListInfo
+} from '../../../utils/string/StringUtils';
 
 const chatAtom = atom({
   key: 'chatAtom',
   default: {
-    selectedChannel: null
+    selectedChannel: null,
+    isLoadingFetchingChannelDetail: false
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of `isLoadingFetchingChannelDetail` to the `chatAtom` default state is a good way to track the loading state when fetching channel details. This can be used to provide feedback to the user about the loading status, improving user experience.
```diff
  const chatAtom = atom({
    key: 'chatAtom',
    default: {
-     selectedChannel: null
+     selectedChannel: null,
+     isLoadingFetchingChannelDetail: false
    }
  });
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->